codegen/template.go

```diff
@@ -637,11 +662,26 @@ func (t *Template) GenerateMainFile(
 	}, nil
 }
 
-func (t *Template) execTemplateAndFmt(templName string, filePath string, data interface{}) error {
+func (t *Template) execTemplateAndFmt(
+	templName string,
+	filePath string,
+	data interface{},
+	copyrightHeader string,
+) error {
 	file, err := openFileOrCreate(filePath)
 	if err != nil {
 		return errors.Wrapf(err, "failed to open file: %s", err)
 	}
+
+	_, err = io.WriteString(file, copyrightHeader)
```

Comment:
It looks like we need to write the "generated" by comments first and then the copyright otherwise it breaks the github UI of collapsing generated code.